src/handler-context/context-builders/index.test.ts

```diff
@@ -0,0 +1,224 @@
+  TestEnvVars,
+  TestConfigFiles
+> = {
+  envVars: ["THIS", "THAT", "THE_OTHER"],
+  messageTypeGuard: (message: any): message is TestMessage =>
+    typeof message.a === "string" &&
+    typeof message.b === "number" &&
+    typeof message.c === "boolean",
+  outputs: [
+    {
+      destination: "THAT",
+      store: mockStoreFunction1,
+    },
+    {
+      destination: "THE_OTHER",
+      store: mockStoreFunction2,
+    },
+  ],
+  configFiles: [ConfigFileNames.inferences, ConfigFileNames.rates],
+};
+
+describe("buildContext", () => {
+  beforeEach(() => {
+    process.env.THIS = "this";
+    process.env.THAT = "that";
+    process.env.THE_OTHER = "the other";
+    process.env.CONFIG_BUCKET = "mock-config-bucket";
+  });
+  afterAll(() => {
+    delete process.env.THIS;
+    delete process.env.THAT;
+    delete process.env.THE_OTHER;
+    delete process.env.CONFIG_BUCKET;
+  });
+
+  it.each([
+    {
+      event: testSQSEvent,
+      expectedMessages: [{ a: "something", b: 12, c: false, _id: "msg_1" }],
+    },
+    { event: testS3Event, expectedMessages: ["test document"] },
+  ])(
+    "Returns a complete handler context based on the given options and event",
+    async ({ event, expectedMessages }) => {
+      const ctx = await buildContext<TestMessage, TestEnvVars, TestConfigFiles>(
+        event,
+        testOptions
+      );
+      // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
+      expect(ctx.logger).toBeInstanceOf(Logger);
+      expect(ctx.env).toEqual({
+        THIS: "this",
+        THAT: "that",
+        THE_OTHER: "the other",
+      });
+      expect(ctx.messages).toEqual(expectedMessages);
+      expect(ctx.outputs[0].destination).toBe("that");
+      expect(ctx.outputs[1].destination).toBe("the other");
+      expect(ctx.config).toEqual({
+        inferences: "mock inferences",
+        rates: "mock rates",
+      });
+    }
+  );
+
+  it("Throws an error if any of the requested env vars are not present", async () => {
+    delete process.env.THE_OTHER;
+    try {
+      await buildContext<TestMessage, TestEnvVars, TestConfigFiles>(
+        testSQSEvent,
+        testOptions
+      );
+    } catch (error) {
+      expect((error as Error).message).toContain("Environment is not valid");
+    }
+    expect.hasAssertions();
+  });
+
+  it("Throws an error if a given message does not conform to the specified type", async () => {
+    try {
+      await buildContext<TestMessage, TestEnvVars, TestConfigFiles>(
+        // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
+        {
+          ...testSQSEvent,
+          Records: [
+            ...testSQSEvent.Records,
+            {
+              messageId: "msg_2",
+              body: `{"doesn't": "conform"}`,
+            },
+          ],
+        } as SQSEvent,
+        testOptions
+      );
+    } catch (error) {
+      expect((error as Error).message).toContain(
+        "Could not process message msg_2"
```

Comment:
This could be a better error message. "Incoming data did not conform to TypeGuard" or similar.